src/v2/api/services/ScheduleService.ts

```diff
@@ -94,4 +74,70 @@ export class ScheduleService {
         return indexOfLesson !== null;
       });
   }
+
+  async getEvent (id: string, week: number) {
+    const { startTime, endTime, ...event } = await this.eventRepository.find({ id });
+    const isEvent = !event.lessons.length;
+
+    week = week ?? await this.dateService.getCurrentWeek();
+
+    const { startDate } = await this.dateService.getCurrentSemester();
+    const startWeek = Math.ceil((startTime.getTime() - startDate.getTime()) / WEEK);
+    const endWeek = Math.ceil((endTime.getTime() - startDate.getTime()) / WEEK);
+
+    startTime.setDate(startTime.getDate() + (week - startWeek) * 7);
+    endTime.setDate(endTime.getDate() - (endWeek - week) * 7);
+
+    if (week > endWeek || week < startWeek) {
+      throw new InvalidWeekException();
+    }
+    if (event.period === Period.EVERY_FORTNIGHT && startWeek % 2 !== week % 2) {
+      throw new InvalidWeekException();
+    }
+
+    const eventInfoIndex = (week - startWeek) / (event.period === Period.EVERY_FORTNIGHT ? 2 : 1);
+    const eventInfo = event.eventInfo[eventInfoIndex].description;
+
+    let teachers = null, disciplineInfo = null, disciplineType: DisciplineType = null;
+    if (!isEvent) {
+      const discipline = await this.disciplineRepository.find({
+        disciplineTypes: {
+          some: {
+            lessons: {
+              some: {
+                eventId: event.id,
+              },
+            },
+          },
+        },
+      });
+      disciplineType = discipline.disciplineTypes.find((dt) => dt.id === event.lessons[0].disciplineTypeId);
+      teachers = discipline.disciplineTeachers
+        .filter((dt) => dt.roles.some((r) => r.disciplineTypeId === disciplineType.id))
+        .map(({ teacher }) => ({
+          id: teacher.id,
+          firstName: teacher.firstName,
+          middleName: teacher.middleName,
+          lastName: teacher.lastName,
+          description: teacher.description,
+          avatar: teacher.avatar,
+          rating: teacher.rating.toNumber(),
```

Comment:
why is that in this case?

Reply:
We do not use the values of these fields, see the design

Reply:
i'm happy for you !!!